### ISSUE TITLE
Make log output from JUnit Platform listeners visible in test report and build scans

### ISSUE THREAD

**Issue by britter** [gradle/gradle#14642]
### Expected Behavior

When implementing a `org.junit.platform.launcher.TestExecutionListener` output written to standard out or standard error should be visible the same way that output from tests is visible in the test report or build scans.

### Current Behavior

Output is only visible on the console when configuring

```groovy
test {
  testLogging {
    events 'STANDARD_OUT', 'STANDARD_ERROR'
  }
}
```

### Context

For test distribution be advice to move setup code for running tests from setup Gradle tasks into JUnit platform listeners (see https://docs.gradle.com/enterprise/test-distribution-gradle-plugin/#provisioning_an_external_resource_for_many_tests). Having log output show up in test reports and build scans is helpful when developing these listeners.

**Comment by britter** [comment]
I think this is still valid. WDYT @marcphilipp?

**Comment by marcphilipp** [comment]
Yes, still valid. I just updated the reproducer linked in the issue description to Gradle 7.4.1.
